pkg/dmaasbackup/dmaasbackup.go

```diff
@@ -33,77 +33,66 @@ import (
 
 	// Delete perform the cleanup required as part of dmaasbackup deletion
 	Delete(obj *v1alpha1.DMaaSBackup, logger logrus.FieldLogger) error
 }
 
 type dmaasBackup struct {
-	velerons       string
-	dmaasClient    clientset.Interface
-	backupLister   velerov1lister.BackupNamespaceLister
-	pvbLister      velerov1lister.PodVolumeBackupLister
-	backupClient   velerov1.BackupInterface
-	scheduleClient velerov1.ScheduleInterface
-	logger         logrus.FieldLogger
-	clock          apimachineryclock.Clock
+	veleroNs           string
+	dmaasClient        clientset.Interface
+	backupLister       velerov1lister.BackupNamespaceLister
+	pvbLister          velerov1lister.PodVolumeBackupLister
+	backupClient       velerov1.BackupInterface
+	scheduleClient     velerov1.ScheduleInterface
+	deleteBackupClient velerov1.DeleteBackupRequestInterface
+	logger             logrus.FieldLogger
+	clock              apimachineryclock.Clock
+
+	// shouldRequeue should be set if immediate reconciliation is needed for object
+	shouldRequeue bool
 }
 
 // NewDMaaSBackupper returns the interface to execute operation on dmaasbackup resource
 func NewDMaaSBackupper(
-	velerons string,
+	veleroNs string,
 	dmaasClient clientset.Interface,
 	veleroClient velerov1.VeleroV1Interface,
 	veleroInformer velerov1informer.Interface,
 	clock apimachineryclock.Clock,
 ) DMaaSBackupper {
 	return &dmaasBackup{
-		velerons:       velerons,
-		dmaasClient:    dmaasClient,
-		backupClient:   veleroClient.Backups(velerons),
-		scheduleClient: veleroClient.Schedules(velerons),
-		clock:          clock,
-		backupLister:   veleroInformer.Backups().Lister().Backups(velerons),
-		pvbLister:      veleroInformer.PodVolumeBackups().Lister(),
+		veleroNs:           veleroNs,
+		dmaasClient:        dmaasClient,
+		backupClient:       veleroClient.Backups(veleroNs),
+		scheduleClient:     veleroClient.Schedules(veleroNs),
+		deleteBackupClient: veleroClient.DeleteBackupRequests(veleroNs),
+		clock:              clock,
+		backupLister:       veleroInformer.Backups().Lister().Backups(veleroNs),
+		pvbLister:          veleroInformer.PodVolumeBackups().Lister(),
+		shouldRequeue:      false,
 	}
 }
 
-func (d *dmaasBackup) Execute(obj *v1alpha1.DMaaSBackup, logger logrus.FieldLogger) error {
+func (d *dmaasBackup) Execute(obj *v1alpha1.DMaaSBackup, logger logrus.FieldLogger) (bool, error) {
 	var err error
 
 	d.logger = logger
 
-	d.logger.Debug("updating schedule information")
+	d.logger.Debug("Executing dmaasbackup")
+	defer d.logger.Debug("Execution of dmaasbackup completed")
 
-	// check for stale velero schedule created by dmaas-operator
-	// This is necessary because there are chances where operator has created
-	// required schedule/backup but missed to update the dmaasbackup object, due to error or restart
-	err = d.updateScheduleInfo(obj)
-	if err != nil {
-		return errors.Wrapf(err, "failed to check schedule information")
-	}
+	// reset shouldRequeue
```

Comment:
better to set this to false in reconciler itself as soon as it got the object..

Reply:
shouldRequeue is having limited scope to dmaasbackup  package only. Caller can unset this value, but i think it will be better if this value can be updated in this package only. Reason is, caller needs to execute extra API to unset shouldRequeue.

Reply:
got it.. this is fine.. my comment is invalid..
I was thinking that caller returns object's value, but, thats a local variable which caller returns..